include/SDL_gpu_shadercross.h

```diff
@@ -0,0 +1,254 @@
+/*
+  Simple DirectMedia Layer Shader Cross Compiler
+  Copyright (C) 2024 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+
+#ifndef SDL_GPU_SHADERCROSS_H
+#define SDL_GPU_SHADERCROSS_H
+
+#ifdef _WIN32
+#define SHADERCROSSAPI __declspec(dllexport)
```

Comment:
Shouldn't we only efine `SHAREDCROSSAPI` to `__declspec(dllexport)` when building a dll library?
We don't want dllexport for people building a static library.

Reply:
tmk on Windows, you still want to put this for static libraries, as on Windows some linkers will refuse to link against symbols that arent exported like this, even if its a static lib